app/main.py

```diff
@@ -1,34 +1,210 @@
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(
+            self,
+            row: int,
+            column: int,
+            is_alive: bool = True
+    ) -> None:
+
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
+
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = self.create_decks()
+
+    def create_decks(self) -> list[Deck]:
+
+        if self.start[0] == self.end[0]:
+            if self.start[1] < self.end[1]:
+                return self.create_decks_instances(
+                    row=self.start[0],
+                    start=self.start[1],
+                    end=self.end[1],
+                    is_row=True
+                )
+
+            if self.start[1] == self.end[1]:
+                return [(Deck(row=self.start[0], column=self.start[1]))]
+
+            return self.create_decks_instances(
+                row=self.start[0],
+                start=self.end[1],
+                end=self.start[1],
+                is_row=True
+            )
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        if self.start[1] == self.end[1]:
+            if self.start[0] < self.end[0]:
+                return self.create_decks_instances(
+                    row=self.start[1],
+                    start=self.start[0],
+                    end=self.end[0]
+                )
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+            return self.create_decks_instances(
+                row=self.start[1],
+                start=self.end[0],
+                end=self.start[0]
+            )
+
+    @staticmethod
+    def create_decks_instances(
+            row: int,
+            start: int,
+            end: int,
+            is_row: bool = False
+    ) -> list[Deck]:
+        decks = []
+        for i in range(start, end + 1):
+            if is_row:
+                decks.append(Deck(row=row, column=i))
+                continue
+            decks.append(Deck(row=i, column=row))
+
+        return decks
+
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+
+    def fire(self, row: int, column: int) -> None:
+        deck = self.get_deck(row, column)
+        deck.is_alive = False
+
+        for deck in self.decks:
+            if deck.is_alive:
+                self.is_drowned = False
+                break
+        else:
```

Comment:
`else` indentation should be in line with `if`.

Reply:
Thanks for your comments.
Here "else" belongs to the "for" statement